packages/backend/src/api/auth/__tests__/hashing.service.test.ts

```diff
@@ -0,0 +1,52 @@
+import * as bcrypt from 'bcrypt';
+import { HashingService } from '../services/hashing.service';
+
+describe('HashingService', () => {
+  let hashingService: HashingService;
+  let genSalt: jest.SpyInstance;
+  let hash: jest.SpyInstance;
+  let compare: jest.SpyInstance;
+
+  beforeAll(async () => {
+    hashingService = new HashingService();
+  });
+
+  beforeEach(async () => {
+    genSalt = jest.spyOn(bcrypt, 'genSalt');
+    hash = jest.spyOn(bcrypt, 'hash');
+    compare = jest.spyOn(bcrypt, 'compare');
+
+    genSalt.mockImplementation((payload: object) => {
+      return payload;
+    });
+    hash.mockImplementation(() => {
+      return 'hashed';
+    });
+
+    compare.mockImplementation((plainPassword: string, password: string) => {
+      if (plainPassword === password) {
+        return true;
+      }
+      return false;
+    });
+  });
+
+  describe('hashPassword', () => {
+    it('should generate encrypted password', async () => {
+      const res = await hashingService.hashPassword('12345678');
+      expect(res).toMatch('hashed');
+    });
+  });
+
+  describe('matchingPassword', () => {
+    it('should return true on matched password', async () => {
+      const res = await hashingService.matchingPassword('12345', '12345');
+      expect(res).toBeTruthy();
```

Comment:
toBeTruthy() accepts anything that is true, for example an non empty string should be true
https://jestjs.io/docs/en/using-matchers#truthiness

```suggestion
      expect(res).toBe(true);
```